dask_kubernetes/operator/controller/controller.py

```diff
@@ -162,40 +162,46 @@ def build_worker_deployment_spec(
     return deployment_spec
 
 
-def get_job_runner_pod_name(job_name):
+def get_job_runner_job_name(job_name):
     return f"{job_name}-runner"
 
 
-def build_job_pod_spec(job_name, cluster_name, namespace, spec, annotations, labels):
+def build_job_spec(job_name, cluster_name, namespace, spec, annotations, labels):
     labels.update(
         **{
             "dask.org/cluster-name": cluster_name,
             "dask.org/component": "job-runner",
             "sidecar.istio.io/inject": "false",
         }
     )
-    pod_spec = {
-        "apiVersion": "v1",
-        "kind": "Pod",
-        "metadata": {
-            "name": get_job_runner_pod_name(job_name),
-            "labels": labels,
-            "annotations": annotations,
+    metadata = {
+        "name": get_job_runner_job_name(job_name),
+        "labels": labels,
+        "annotations": annotations,
+    }
+    job_spec = {
+        "apiVersion": "batch/v1",
+        "kind": "Job",
+        "metadata": metadata,
+        "spec": {
```

Comment:
I think we probably want our spec to replace this, rather than the `"spec"` within `"template"`.

That does mean we need to modify our CRDs to ref a `JobSpec` instead. That is a breaking change to our CRDs so we might need to think about how to bump the version there.

https://github.com/dask/dask-kubernetes/blob/2aa43eda392e8de899993ca019bbf1da11b6c303/dask_kubernetes/operator/customresources/daskjob.yaml#L65